src/app/containers/MostRead/Amp/index.jsx

```diff
@@ -0,0 +1,144 @@
+import React, { useContext } from 'react';
+import { Helmet } from 'react-helmet';
+import { string, oneOf, elementType } from 'prop-types';
+import styled from '@emotion/styled';
+import {
+  AMP_LIST_JS,
+  AMP_MUSTACHE_JS,
+  AMP_SCRIPT_JS,
+} from '@bbc/psammead-assets/amp-boilerplate';
+import { ServiceContext } from '#contexts/ServiceContext';
+import { getSansRegular } from '@bbc/psammead-styles/font-styles';
+import pathOr from 'ramda/src/pathOr';
+import { C_SHADOW } from '@bbc/psammead-styles/colours';
+import { getBodyCopy } from '@bbc/gel-foundations/dist/typography';
+import { MostReadItemWrapper, MostReadLink } from '../Canonical/Item';
+import MostReadRank, { serviceNumerals } from '../Canonical/Rank';
+import generateCSPHash from '../utilities/generateCPSHash';
+
+const rankTranslationScript = (endpoint, service) => {
+  const translation = serviceNumerals(service);
+  return `
+  const translations = ${JSON.stringify(translation)}
+  const getRemoteData = async () => {
+    try{
+      const response = await fetch("${endpoint}");
+      const data = await response.json();
+      data.records.forEach((item, index) => {
+        return item.rankTranslation = translations[index+1]
+      });
+      return data;
+    } catch(error){
+      document.body.removeChild(document.body.firstElementChild)
+      return [];
+    }
+  }
+    exportFunction('getRemoteData', getRemoteData);`;
+};
+
+const AmpMostRead = ({ endpoint, size, wrapper: Wrapper }) => {
+  const {
+    service,
+    script,
+    dir,
+    mostRead: { numberOfItems },
+    translations,
+  } = useContext(ServiceContext);
+
+  const onlyinnerscript = rankTranslationScript(endpoint, service);
+
+  const FallbackText = styled.p`
+    ${() => getSansRegular(service)}
+    ${() => getBodyCopy(script)}
+    color: ${C_SHADOW};
+    margin: 0;
+  `;
+
+  const fallbackText = pathOr(
+    'Content is not available',
+    ['socialEmbed', 'fallback', 'text'],
+    translations,
+  );
+
+  return (
+    <amp-script id="dataFunctions" script="local-script">
+      <Wrapper>
+        <Helmet
+          script={[
+            {
+              id: 'local-script',
+              type: 'text/plain',
+              target: 'amp-script',
+              innerHTML: onlyinnerscript,
+            },
+          ]}
+        />
+        <Helmet>
+          {/* Import required amp scripts for most read */}
+          {AMP_LIST_JS}
+          {AMP_MUSTACHE_JS}
+          {AMP_SCRIPT_JS}
+          <meta
+            name="amp-script-src"
+            content={generateCSPHash(
+              onlyinnerscript,
+              'sha384',
+              'utf8',
+              'base64',
```

Comment:
Looks great. Keen to know the performance impact (if any) on running the hash generation each time.

Reply:
Yh we were curiously with that too but not too sure how to measure it.